tensorbay/exception.py

```diff
@@ -72,8 +81,10 @@ class DatasetTypeError(ClientError):
 
     """
 
-    def __init__(self, dataset_name: str, is_fusion: bool) -> None:
-        super().__init__()
+    def __init__(
+        self, message: Optional[str] = None, *, dataset_name: str, is_fusion: bool
```

Comment:
Should the keyword arguments have default values?
Also in `__str__`, what if the exception is initialized with `message`?
Same for other exceptions.